src/Microsoft.DocAsCode.Build.Common/YamlHtmlPart.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.Common
+{
+    using System.Collections.Generic;
+
+    using HtmlAgilityPack;
+
+    public class YamlHtmlPart
+    {
+        public HtmlDocument Doc { get; set; }
+        public int StartLine { get; set; }
+        public int EndLine { get; set; }
+
+        public static IList<YamlHtmlPart> SplitYamlHtml(string html)
+        {
+            var doc = new HtmlDocument();
+            doc.LoadHtml(html);
+            var parts = new List<YamlHtmlPart>();
+
+            var nodes = doc.DocumentNode.SelectNodes("//yamlheader");
+            if (nodes == null)
+            {
+                return parts;
+            }
+
+            foreach (var node in nodes)
+            {
+                var part = new HtmlDocument();
+
+                var startLineStr = node.GetAttributeValue("start", "-1");
+                var endLineStr = node.GetAttributeValue("end", "-1");
+
+                int startLine, endLine;
+                if (!int.TryParse(startLineStr, out startLine))
+                {
+                    startLine = -1;
+                }
+                if (!int.TryParse(endLineStr, out endLine))
+                {
+                    endLine = -1;
+                }
+
+                var currentNode = node;
+
+                do
+                {
+                    var nextNode = currentNode.NextSibling;
+                    part.DocumentNode.AppendChild(currentNode);
+                    currentNode = nextNode;
```

Comment:
The `AppendChild` method has side effect that would change the status of `currentNode`

e.g. 

```csharp
var expectNext = node.NextSibling;
part.DocumentNode.AppendChild(node);
var actualNext = node.NextSibling;
```
`expectNext` is not the same as `actualNext`